### PR TITLE
dev-tim

### PR DESCRIPTION
**Pull request by timomer**
Added ServiceClientConnaction and MainActivity clean up

### CODE REVIEW

**Comment by timomer** [comment]
Hey @GeoffWyche, this is a bigger update that you deff want to review. In summary...

Added the ServiceClientConnection.class, an object of this class is created and used for all comms from the app to the RT2 Sevice, be it from UI or Communication Service I will add later. This is the only connection to RT2 Sevice.

Updated MainActivity to use ServiceClientConnectionobject, changed how broadcast listeners are registered and moved the group of IF statements to SWITCH as easier to read. 

Note that Pump ID and RL Address are now read from Prefs, so you will need to add them in the settings menu.

File: app/src/main/java/com/gxwtech/roundtrip2/RoundtripService/RoundtripService.java [hunk 1]
```diff
@@ -86,11 +87,17 @@ public void onCreate() {
         mContext = getApplicationContext();
         serviceConnection = new RoundtripServiceIPCConnection(mContext);
 
-        sharedPref = mContext.getSharedPreferences(RT2Const.serviceLocal.sharedPreferencesKey, Context.MODE_PRIVATE);
```
Comment (timomer) [inline app/src/main/java/com/gxwtech/roundtrip2/RoundtripService/RoundtripService.java:89]: Is this cool? From what I have read there is no about Context.MODE_PRIVATE there is no difference using using it or not

**Comment by timomer** [comment]
FYI - with this most recent code, I get "Pump: Lost", please test to see if you get this as well. If not ill provide logs once this PR is accepted.

File: app/src/main/java/com/gxwtech/roundtrip2/RoundtripService/RoundtripService.java (hunk 1 above)
Comment (GeoffWyche) [inline app/src/main/java/com/gxwtech/roundtrip2/RoundtripService/RoundtripService.java:89]: Sure, sounds good.  I'm still pretty new at the Android libraries.

**Comment by GeoffWyche** [comment]
Yeah, I still get "Pump: Lost" a bunch.  It just means that the first time it tried to read the pump model number it didn't work right.  Haven't tracked down why.  Most of the time the other pump operations work just fine.

**Pull request merged**